src/components/menu/utils/RenderChildrenInPortal.tsx

```diff
@@ -0,0 +1,38 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+export interface IRenderChildrenInPortalProps {
+    targetElement: Element;
+    children: React.ReactNode;
+}
+
+export default class RenderChildrenInPortal extends React.Component<IRenderChildrenInPortalProps> {
+    private portalContentWrapperEl: HTMLElement;
+
+    private constructor(props: IRenderChildrenInPortalProps) {
+        super(props);
+
+        const wrapperEl = document.createElement('div');
+        this.portalContentWrapperEl = wrapperEl;
+    }
+
+    public componentWillMount() {
```

Comment:
componentDidMount

Reply:
Actually this one needs to be componentWillMount because we want it to mount to the body (or whatever is in portalTarget) before rendering the component. When I changed it to componentDidMount, elements has 0 width/height on first render, because they were only mounted to some div that was not attached to the document.